File: typescript/src/schemas/cldpm.ts
/**
 * CLDPM root configuration schema.
 */

import { z } from "zod";

/**
 * Schema for cldpm.json configuration file.
 */
export const CldpmConfigSchema = z.object({
  name: z.string().min(1, "Name is required"),
  version: z.string().default("1.0.0"),
  projectsDir: z.string().default("projects"),
  sharedDir: z.string().default("shared"),
});

export type CldpmConfig = z.infer<typeof CldpmConfigSchema>;

/**
 * Create a CldpmConfig with defaults.
 */
export function createCldpmConfig(
  name: string,
  options: Partial<Omit<CldpmConfig, "name">> = {}
): CldpmConfig {
  return CldpmConfigSchema.parse({
    name,
    ...options,
  });
}

File: python/cldpm/schemas/cldpm.py
"""Pydantic model for cldpm.json (root configuration)."""

from pydantic import BaseModel, Field


class CldpmConfig(BaseModel):
    """Root configuration for a CLDPM mono repo."""

    name: str = Field(..., description="Name of the mono repo")
    version: str = Field(default="1.0.0", description="Version of the mono repo")
    projects_dir: str = Field(
        default="projects",
        alias="projectsDir",
        description="Directory containing projects",
    )
    shared_dir: str = Field(
        default="shared",
        alias="sharedDir",
        description="Directory containing shared components",
    )

    model_config = {
        "populate_by_name": True,
        "json_schema_extra": {
            "example": {
                "name": "my-monorepo",
                "version": "1.0.0",
                "projectsDir": "projects",
                "sharedDir": "shared",
            }
        },
    }
